test/jdk/java/awt/Window/ProxyCrash/PopupProxyCrash.java

```diff
@@ -0,0 +1,192 @@
+    }
+
+    private static void createUI() {
+        frame = new Frame("PopupProxyCrash");
+        Font dialog = new Font("Dialog", Font.PLAIN, 12);
+        Font serif = new Font("Serif", Font.PLAIN, 12);
+        Font monospaced = new Font("Monospaced", Font.PLAIN, 12);
+
+        fonts = new Font[] { dialog, serif, monospaced };
+
+        cb = new JComboBox(fonts);
+
+        cb.setLightWeightPopupEnabled(false);
+        comboBoxUI = new MyComboBoxUI();
+        cb.setUI(comboBoxUI);
+        jtf = new JTextField("JTextField");
+        jtf.setFont(fonts[1]);
+        tf = new Button("TextField");
+        tf.setFont(fonts[1]);
+        cb.addActionListener(new PopupProxyCrash());
+
+        panel = new Panel() {
+            public Dimension getPreferredSize() {
+                return new Dimension(100, 20);
+            }
+            public void paint(Graphics g) {
+                System.out.println("Painting with font " + getFont());
+                g.setColor(Color.white);
+                g.fillRect(0, 0, getWidth(), getHeight());
+                g.setColor(Color.black);
+                g.setFont(getFont());
+                g.drawString("LightWeight", 10, 10);
+            }
+        };
+        panel.setFont(fonts[1]);
+
+        Container parent = Box.createVerticalBox();
+        parent.add(jtf);
+        parent.add(tf);
+        parent.add(panel);
+        parent.add(cb);
+
+        frame.add(parent, BorderLayout.CENTER);
+        frame.setSize(200,200);
+        frame.pack();
+        frame.setLocationRelativeTo(null);
+        frame.setVisible(true);
+    }
+
+    private static Point getComboBoxLocation() throws Exception {
+        final Point[] result = new Point[1];
+
+        EventQueue.invokeAndWait(() -> {
+            Point point = cb.getLocationOnScreen();
+            Dimension size = cb.getSize();
+
+            point.x += size.width / 2;
+            point.y += size.height / 2;
+            result[0] = point;
+        });
+        return result[0];
+    }
+
+    private static Point getItemPointToClick(final int item) throws Exception {
+        final Point[] result = new Point[1];
+
+        EventQueue.invokeAndWait(() -> {
+            BasicComboPopup popup = (BasicComboPopup)comboBoxUI.getComboPopup();
+            Point point = popup.getLocationOnScreen();
+            Dimension size = popup.getSize();
+
+            int step = size.height / fonts.length;
+            point.x += size.width / 2;
+            point.y += step / 2 + step * item;
+            result[0] = point;
+        });
+        return result[0];
+    }
+
+    static void runTest() throws Exception {
+        for (int i = 0; i < TEST_COUNT; i++) {
+            Point point = getComboBoxLocation();
+            robot.mouseMove(point.x, point.y);
+            robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+            robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+            robot.delay(500);
+            robot.waitForIdle();
```

Comment:
Seen many review comments where it is advised to invoke delay after waitForIdle.

Reply:
Updated.